relay-server/src/actors/spooler/mod.rs

```diff
@@ -501,18 +394,330 @@ impl BufferService {
         }
     }
 
-    /// Checks if the spool still has space and sets `is_disk_full` flag accordingly.
-    async fn refresh_spool_state(&mut self) {
-        let Some(BufferSpoolConfig {
+    /// Dequeues the envelopes from the on-disk spool and send them to the provided `sender`.
+    ///
+    /// The keys for which the envelopes could not be fetched, send back to `ProjectCache` to merge
+    /// back into index.
+    async fn dequeue(
+        &self,
+        project_key: ProjectKey,
+        keys: &mut Vec<QueueKey>,
+        sender: mpsc::UnboundedSender<ManagedEnvelope>,
+        services: &Services,
```

Comment:
Is there any reason why `services` can't be provided to the buffer, so that we can remove this parameter?

Reply:
Only that it used in one place, which makes it easier to pass to function in one place only.